### PR TITLE
Support memcached for storing execution payloads, allow disabling database

### PR DESCRIPTION
**Pull request by metachris**
## 📝 Summary

Saving execution payloads in the database takes terrabytes of data, and they are not automatically truncated. This PR adds memcached as another data redundancy and allows disabling saving the execution payloads in the database.

Continuation of the work https://github.com/Wazzymandias has done in

* #303 

In particular:

* memcache fix: 3ddd53039ac16eb93eea40d1b122d2ee41be3732
* [disable saving execution payload to database](https://github.com/flashbots/mev-boost-relay/commit/a69216298357780c4ce53369d958d4e1b8bf4c04)

Has a new env var: `DISABLE_PAYLOAD_DATABASE_STORAGE`

Tested in Zhejiang: https://boost-relay-zhejiang.flashbots.net/

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

**Comment by metachris** [comment]
Confirming that blocks received through memcache works, by disabling both database and redis in Zhejiang!

This PR is ready to get merged.

File: datastore/datastore.go [hunk 1]
```diff
@@ -127,24 +129,36 @@ func (ds *Datastore) GetGetPayloadResponse(slot uint64, proposerPubkey, blockHas
...
 
-	// 2. try to get from database
-	blockSubEntry, err := ds.db.GetExecutionPayloadEntryBySlotPkHash(slot, proposerPubkey, blockHash)
+	// 2. try to get from Memcached
+	if ds.memcached != nil {
+		resp, err = ds.memcached.GetExecutionPayload(slot, _proposerPubkey, _blockHash)
+		if err != nil {
+			ds.log.WithError(err).Error("error getting execution payload response from memcached")
+		} else if resp != nil {
+			ds.log.Debug("getPaylaod response from memcached")
```
Comment (michaelneuder) [inline datastore/datastore.go:138]: typo: s/getPaylaod/getPayload

**Review by michaelneuder — COMMENTED** [review]
LGTM, i like the redundancy, and i think it would benefit all relays to avoid saving all the execution payloads :-)

File: datastore/datastore.go (hunk 1 above)
Comment (metachris) [reply datastore/datastore.go]: great catch, fixed!

**Pull request merged**